### DISCUSSION TITLE
Preact HTM vs JSX

### DISCUSSION THREAD

**Discussion by ThePeach** [freshframework/fresh#543] [Q&A] [answered]
Hi all.

Would be there any major difference in [using HTM](https://preactjs.com/guide/v10/getting-started#alternatives-to-jsx) with Fresh instead of JSX? Couldn't find any resource that picked up on that. If so, is there any adjustment needed to start using it? 

Thanks for any hints

**Comment by marvinhagemeister** [comment, marked as answer]
The main difference of JSX vs `htm` is that JSX is type safe, but needs some form of transpilation to turn it into valid JavaScript. `htm` on the other hand already is valid JavaScript and can be used as is in the browser. It's meant as an alternative to JSX. The downside is that it cannot be made type safe as extending TypeScript is very tricking and they don't have a proper plugin API for enhancing type checking so far.

**Reply by ThePeach** [reply]
thanks a lot for the explanation!
